### PR TITLE
Update ESLint check to align with Express

### LINKED ISSUE #32: Update ESLint required for declarations in orm and controller layers
Unavoidable cases like these will fail the *ESLint* check, even though this follow *Express* documentation:

1. Function declaration, following *Express* module export (e.g. `addPendingMatch: addPendingMatch` will fail the check)
```javascript
const pendingMatchController = {
   addPendingMatch: addPendingMatch,
   deleteByUsername: deleteByUsername,
};
```
2. Required *Express* function pre-declaration will fail check as functions are declared before definition (e.g. `addPendingMatch` is used before function definition)
```javascript
const pendingMatchController = {
   addPendingMatch: addPendingMatch,
   deleteByUsername: deleteByUsername,
};
 
function addPendingMatch(req, res) {
   const pendingMatch = req.body;
   pendingMatchOrm.create(pendingMatch).then(
       (data) => {
           res.send(data);
       },
   ).catch((err) => {
       console.log(err);
   });
}
```

### PR DESCRIPTION
**Pull request by Punpun1643**
Fixes #32 

Proposed commit message:
```
ESLint check fails due to some check that
does not align with Express.

An update is required to ignore these checks.

Include object-shorthand:0 to ignore necessary
declarations for Express module exports.

Include no-use-before-define to ignore functions
as these need to be `used` before declarations
following Express.

Let's update ESLint to align with these requirements.
```

Changes made:
1. Modify checks to by pass function use before declarations as these are required to easy module export later following *Express* documentation. Example:
```javascript
const pendingMatchController = {
   addPendingMatch: addPendingMatch,
   deleteByUsername: deleteByUsername,
};
 
function addPendingMatch(req, res) {
   const pendingMatch = req.body;
   pendingMatchOrm.create(pendingMatch).then(
       (data) => {
           res.send(data);
       },
   ).catch((err) => {
       console.log(err);
   });
}
```
2. Update ESLint to ignore object shorthand required for *Express* module export. Example:
```javascript
const pendingMatchController = {
   addPendingMatch: addPendingMatch,
   deleteByUsername: deleteByUsername,
};
```

### CODE REVIEW

**Review by fantablack — APPROVED** [review]
Looks good! Necessary fix for a valid linting problem. LGTM!

File: .eslintrc.json [hunk 1]
```diff
@@ -34,6 +34,14 @@
...
         ],
 
+        "object-shorthand": 0,
+        "no-use-before-define": [
+            "error",
+            {
+                "functions": false
+            }
+        ],
+
```
Comment (fantablack) [inline .eslintrc.json:44]: Resolved merge conflict and reformatted rules

**Review by fantablack — APPROVED** [review]
Resolve merge conflict and reformatted rules

**Review by dionegoh — APPROVED** [review]
LGTM! :)

**Pull request merged**